--- a/internal/connection/pool.go
+++ b/internal/connection/pool.go
@@ -39,6 +39,12 @@
 func (p *ConnPool[T]) Get(id string) (T, error) {
 	p.m.Lock()
 	defer p.m.Unlock()
+	if cached, ok := p.conns[id]; ok {
+        if state, supports := any(cached).(interface{ IsClosed() bool }); supports && state.IsClosed() {
+            delete(p.conns, id)
+            cached.Close()
+        }
+    }
 	if t, ok := p.conns[id]; !ok {
 		err := p.refreshLocked(id)
 		if err != nil {
--- a/internal/nui/kv.go
+++ b/internal/nui/kv.go
@@ -70,7 +70,7 @@
 	var statuses []BucketState
 	for status := range kvLister.Status() {
 		if kvLister.Error() != nil {
-			return a.logAndFiberError(c, err, 500)
+			return a.logAndFiberError(c, kvLister.Error(), 500)
 		}
 		statuses = append(statuses, NewBucketState(status, nil))
 	}
@@ -90,7 +90,7 @@
 	// retrieve config from the related stream cause no direct api available for get bucket config
 	stream, ok, err := a.streamOrFail(c, KV_STREAM_PREFIX+c.Params("bucket"))
 	if !ok {
-		return a.logAndFiberError(c, err, 500)
+		return err
 	}
 	streamInfo, err := stream.Info(c.Context())
 	if err != nil {
@@ -122,7 +122,7 @@
 func (a *App) HandleCreateBucket(c *fiber.Ctx) error {
 	js, ok, err := a.jsOrFail(c)
 	if !ok {
-		return a.logAndFiberError(c, err, 500)
+		return err
 	}
 	bucketConfig := jetstream.KeyValueConfig{}
 	if err := c.BodyParser(&bucketConfig); err != nil {
@@ -142,12 +142,12 @@
 func (a *App) HandleUpdateBucket(c *fiber.Ctx) error {
 	js, ok, err := a.jsOrFail(c)
 	if !ok {
-		return a.logAndFiberError(c, err, 500)
+		return err
 	}
 	bucketName := c.Params("bucket")
 	_, ok, err = a.bucketOrFail(c, bucketName)
 	if !ok {
-		return a.logAndFiberError(c, err, 500)
+		return err
 	}
 	bucketConfig := jetstream.KeyValueConfig{}
 
@@ -170,11 +170,11 @@
 func (a *App) HandleDeleteBucket(c *fiber.Ctx) error {
 	js, ok, err := a.jsOrFail(c)
 	if !ok {
-		return a.logAndFiberError(c, err, 500)
+		return err
 	}
 	_, ok, err = a.bucketOrFail(c, c.Params("bucket"))
 	if !ok {
-		return a.logAndFiberError(c, err, 500)
+		return err
 	}
 	err = js.DeleteKeyValue(c.Context(), c.Params("bucket"))
 	if err != nil {
@@ -186,9 +186,13 @@
 func (a *App) HandleIndexKeys(c *fiber.Ctx) error {
 	kv, ok, err := a.bucketOrFail(c, c.Params("bucket"))
 	if !ok {
-		return a.logAndFiberError(c, err, 500)
+		return err
 	}
 	watcher, err := kv.WatchAll(c.Context(), jetstream.MetaOnly())
+    if err != nil {
+        return a.logAndFiberError(c, err, 500)
+    }
+    defer watcher.Stop()
 
 	keysData := make([]KevValueEntry, 0)
 
@@ -222,7 +226,7 @@
 func (a *App) HandlePurgeDeletedKeys(c *fiber.Ctx) error {
 	kv, ok, err := a.bucketOrFail(c, c.Params("bucket"))
 	if !ok {
-		return a.logAndFiberError(c, err, 404)
+		return err
 	}
 	err = kv.PurgeDeletes(c.Context(), jetstream.DeleteMarkersOlderThan(-1))
 	if err != nil {
